tst/EnergyPlus/unit/InputProcessor.unit.cc

```diff
@@ -74,6 +74,25 @@ namespace EnergyPlus {
 
 		}
 
+		TEST_F(InputProcessorFixture, MisleadingIDDWarningTest)
+		{
+			bool errors_found = false;
+
+			std::string const idf_contents = delimited_string({
+				"Timestep, 4;"
+			});
+
+			ASSERT_FALSE( process_idf( idf_contents, errors_found ) );
+			EXPECT_TRUE( compare_err_stream( "", true ) );
+
+			std::string saveMatchVersion = DataStringGlobals::MatchVersion;
+			DataStringGlobals::MatchVersion = "2.0";
+			ASSERT_FALSE(process_idf(idf_contents, errors_found ));
+			EXPECT_TRUE( compare_err_stream( "   ** Severe  ** IP: Possible incorrect IDD File\r\n   **   ~~~   ** " + DataStringGlobals::IDDVerString +" not the same as expected =\"2.0\"\r\n", true ) );
```

Comment:
@mjwitte Build the string with `delimited_string()` like you do for idf_contents. That function takes care of line endings on the different operating systems. 

I prefer long (and exact) comparison strings because it is more explicit and will show failures if anything in the content is changed in the future.

Also, this is still just testing the `EnergyPlusFixture::process_idf()` function and not `ProcessInput()`. I don't know if this is an issue or not.